metals/src/main/scala/scala/meta/internal/metals/codeactions/ActionableDiagnostic.scala

```diff
@@ -0,0 +1,62 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals._
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class ActionableDiagnostic() extends CodeAction {
+  override def kind: String = l.CodeActionKind.QuickFix
+
+  override def contribute(
+      params: l.CodeActionParams,
+      token: CancelToken,
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+    def path =
+      params.getTextDocument.getUri.toAbsolutePath.toURI.toString
+
+    def createDiagnosticAction(
+        diagnostic: l.Diagnostic,
+        textEdit: l.TextEdit,
+    ): l.CodeAction = {
+      val codeAction = new l.CodeAction()
+      codeAction.setKind(l.CodeActionKind.QuickFix)
+      codeAction.setTitle(ActionableDiagnostic.title)
+      codeAction.setDiagnostics(List(diagnostic).asJava)
+      codeAction.setEdit(
+        new l.WorkspaceEdit(
+          Map(path -> List(textEdit).asJava).asJava
+        )
+      )
+      codeAction
+    }
+
+    val codeActions = params
+      .getContext()
+      .getDiagnostics()
+      .asScala
+      .groupBy {
+        case ScalacDiagnostic.ScalaAction(textEdit) =>
+          Some(textEdit)
+        case _ => None
+      }
+      .collect {
+        case (Some(textEdit), diags)
+            if params.getRange().overlapsWith(diags.head.getRange()) =>
+          createDiagnosticAction(diags.head, textEdit)
+      }
+      .toList
+      .sorted
+
+    Future.successful(codeActions)
+  }
+}
+
+object ActionableDiagnostic {
+  def title = "Apply suggestion"
```

Comment:
I commented on it in the tests, but the diagnostic message mention what the applied suggestion will be?

Reply:
For now no, so I updated it and use [AnnotatedTextEdit](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#annotatedTextEdit) to send some more additional message about the applied suggestion.  I will update this PR after release `scala-cli` to `0.1.17`.

For dependency actionable diagnostic - in field `AnnotatedTextEdit. annotationId` I set value to: `update dependency to 0.8.1`.

<img width="523" alt="Screenshot 2022-10-17 at 11 55 41" src="https://user-images.githubusercontent.com/46607934/196183317-d380eed2-77b0-40e3-9711-14be89cec9dc.png">

Reply:
Annotated text edit seems a bit more complex and might not be supported by all the clients, what about doing:

`Fix: <diagnostic message>` truncated to a sensible length?